edb/load_ext/main.py

```diff
@@ -123,12 +125,16 @@ def install_edgedb_extension(
 
         for f in files:
             target_file = target / f
+            ttarget_file = ttarget / f
 
             with z.open(f) as src:
-                with open(target_file, "wb") as dst:
+                with open(ttarget_file, "wb") as dst:
                     print("Installing", target_file)
                     shutil.copyfileobj(src, dst)
 
+        os.makedirs(ext_dir, exist_ok=True)
+        shutil.move(ttarget, ext_dir)
```

Comment:
I assumed that failing in that case was *why* you didn't like rmtree. Otherwise I don't really understand why we are making this change?

Reply:
Failing if already exists is OK as long as we raise a slightly better error message (e.g. "extension foo is already installed").  The initial `rmtree` concern was that we were removing things permanently _before_ installing new stuff rather than _after_ once we've been satisfied that the extension was installed correctly.